dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/SampleToBatchPadding.scala

```diff
@@ -0,0 +1,184 @@
+ * Intel Corporation licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.dataset.text
+
+import java.util
+
+import com.intel.analytics.bigdl.dataset._
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.tensor.{DoubleType, FloatType, Storage, Tensor}
+
+import scala.collection.Iterator
+import scala.reflect.ClassTag
+
+object SampleToBatchPadding {
+  def apply[T: ClassTag]
+  (totalBatch : Int,
+   padFeature : Boolean = false,
+   padLabel : Boolean = false,
+   padFeatureValue : T = 0,
+   padLabelValue : T = 0,
+   padFeatureDim : Int = 1,
+   padLabelDim : Int = 1,
+   fixLength: Option[Int] = None)
+  (implicit ev: TensorNumeric[T]): SampleToBatchPadding[T]
+  = new SampleToBatchPadding[T](totalBatch, padFeature, padLabel,
+    padFeatureValue, padLabelValue, padFeatureDim, padLabelDim, fixLength)
+}
+
+/**
+ * convert sample to MiniBatch, and you can choose padding feature or label
+ * to same length in a batch
+ * @param totalBatch batchsize
+ * @param padFeature whether padding feature to the same length in a batch
+ * @param padLabel whether padding label to the same length in a batch
+ * @param padFeatureValue  padding feature value, default 0
+ * @param padLabelValue padding label value, default 0
+ * @param padFeatureDim padding feature dim, default dimension 1
+ * @param padLabelDim padding label dim, default dimension 1
+ * @param fixLength padding fix length, default the max first dimension length of feature
+ *                  or label in a batch
+ */
+class SampleToBatchPadding[T: ClassTag]
+(totalBatch : Int,
+ padFeature : Boolean,
+ padLabel : Boolean,
+ padFeatureValue : T = 0,
+ padLabelValue : T = 0,
+ padFeatureDim : Int = 1,
+ padLabelDim : Int = 1,
+ fixLength: Option[Int] = None)
+(implicit ev: TensorNumeric[T])
+  extends Transformer[Sample[T], MiniBatch[T]] {
+
+  private def padding(data: Array[T], padValue: T): Unit = {
+    ev.getType() match {
+      case DoubleType =>
+        util.Arrays.fill(data.asInstanceOf[Array[Double]], 0, data.length,
+          ev.toType[Double](padValue))
+      case FloatType =>
+        util.Arrays.fill(data.asInstanceOf[Array[Float]], 0, data.length,
+          ev.toType[Float](padValue))
+      case _ => throw new UnsupportedOperationException(
+        "SampleToBatchPadding: Only Float/Double supported")
+    }
+  }
```

Comment:
How about providing the (fromIndex and toIndex) parameters for Arrays.fill to avoid filling the whole array each time?

def padding(data: Array[T], fromIndex: Int, toIndex: Int, padValue: T)